From 756645a8c685651c4876330aa2ef8ad583c7753e Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Wed, 3 Apr 2024 14:37:53 +0200
Subject: [PATCH 0749/1695] plat-stm32mp2: conf: enable SCP DVFS

Enable SCMI performance in SCP firmware for STM32MP2.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 5ccf48cfd9337cc2bf7329b4e0d5126a5b8bc378 plat-stm32mp2: conf: enable SCP DVFS]
Change-Id: I319e7075abd09fbd5ea987fc09e7fb88f70bbf19
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/371772
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/lib/scmi-server/conf-optee-stm32mp2.mk | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/core/lib/scmi-server/conf-optee-stm32mp2.mk b/core/lib/scmi-server/conf-optee-stm32mp2.mk
index 686702972..371b308e3 100644
--- a/core/lib/scmi-server/conf-optee-stm32mp2.mk
+++ b/core/lib/scmi-server/conf-optee-stm32mp2.mk
@@ -1,20 +1,21 @@
 $(call force,CFG_SCPFW_MOD_CLOCK,y)
-$(call force,CFG_SCPFW_MOD_DVFS,n)
+$(call force,CFG_SCPFW_MOD_DVFS,y)
 $(call force,CFG_SCPFW_MOD_MSG_SMT,y)
 $(call force,CFG_SCPFW_MOD_OPTEE_CLOCK,y)
 $(call force,CFG_SCPFW_MOD_OPTEE_CONSOLE,y)
 $(call force,CFG_SCPFW_MOD_OPTEE_MBX,y)
 $(call force,CFG_SCPFW_MOD_OPTEE_RESET,y)
 $(call force,CFG_SCPFW_MOD_OPTEE_VOLTD_REGULATOR,y)
-$(call force,CFG_SCPFW_MOD_PSU,n)
-$(call force,CFG_SCPFW_MOD_PSU_OPTEE_REGULATOR,n)
+$(call force,CFG_SCPFW_MOD_PSU,y)
+$(call force,CFG_SCPFW_MOD_PSU_OPTEE_REGULATOR,y)
 $(call force,CFG_SCPFW_MOD_RESET_DOMAIN,y)
 $(call force,CFG_SCPFW_MOD_SCMI,y)
 $(call force,CFG_SCPFW_MOD_SCMI_CLOCK,y)
-$(call force,CFG_SCPFW_MOD_SCMI_PERF,n)
+$(call force,CFG_SCPFW_MOD_SCMI_PERF,y)
 $(call force,CFG_SCPFW_MOD_SCMI_RESET_DOMAIN,y)
 $(call force,CFG_SCPFW_MOD_SCMI_VOLTAGE_DOMAIN,y)
 $(call force,CFG_SCPFW_MOD_VOLTAGE_DOMAIN,y)
+$(call force,CFG_SCPFW_SCMI_PERF_PROTOCOL_OPS,y)
 
 # Info level is sufficient for scp-firmware
 CFG_SCPFW_LOG_LEVEL ?= 1
-- 
2.47.2

